src/WebJobs.Script.WebHost/Diagnostics/SystemLogger.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Azure.WebJobs.Logging;
+using Microsoft.Azure.WebJobs.Script.Config;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
+{
+    public class SystemLogger : ILogger
+    {
+        private readonly bool _isFunctionUserCategory = false;
+        private readonly IEventGenerator _eventGenerator;
+        private readonly ScriptSettingsManager _settingsManager;
+        private readonly string _appName;
+        private readonly string _subscriptionId;
+        private readonly string _categoryName;
+        private readonly string _functionName;
+
+        public SystemLogger(string categoryName, IEventGenerator eventGenerator, ScriptSettingsManager settingsManager)
+        {
+            _settingsManager = settingsManager;
+            _appName = _settingsManager.AzureWebsiteUniqueSlotName;
+            _subscriptionId = Utility.GetSubscriptionId();
+            _eventGenerator = eventGenerator;
+            _categoryName = categoryName;
+            _isFunctionUserCategory = LogCategories.IsFunctionUserCategory(categoryName);
+            _functionName = LogCategories.IsFunctionCategory(_categoryName) ? _categoryName.Split('.')[1] : string.Empty;
+        }
+
+        public IDisposable BeginScope<TState>(TState state) => DictionaryLoggerScope.Push(state);
+
+        public bool IsEnabled(LogLevel logLevel)
+        {
+            // Never log a message that came from the User logs.
+            return !_isFunctionUserCategory;
+        }
+
+        public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
+        {
+            if (!IsEnabled(logLevel))
+            {
+                return;
+            }
+
+            string formattedMessage = formatter?.Invoke(state, exception);
+
+            // If we don't have a message, there's nothing to log.
+            if (string.IsNullOrEmpty(formattedMessage))
+            {
+                return;
+            }
+
+            // Apply standard event properties
+            // Note: we must be sure to default any null values to empty string
+            // otherwise the ETW event will fail to be persisted (silently)
+            string subscriptionId = _subscriptionId ?? string.Empty;
+            string appName = _appName ?? string.Empty;
+            string source = _categoryName ?? string.Empty;
+            string summary = Sanitizer.Sanitize(formattedMessage) ?? string.Empty;
+
+            // eventName is not currently used
```

Comment:
Hmmm, we need to figure this out. My recollection was that the metrics logger was passing custom event names through via that property